osgp/platform/osgp-adapter-domain-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/domain/da/infra/jms/core/messageprocessors/GetDataResponseMessageProcessor.java

```diff
@@ -0,0 +1,105 @@
+import org.opensmartgridplatform.shared.infra.jms.NotificationResponseMessageSender;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessage;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessageResultType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component("domainDistributionAutomationGetDataResponseMessageProcessor")
+public class GetDataResponseMessageProcessor extends BaseNotificationMessageProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetDataResponseMessageProcessor.class);
+
+    @Autowired
+    @Qualifier("domainDistributionAutomationAdHocManagementService")
+    private AdHocManagementService adHocManagementService;
+
+    @Autowired
+    protected GetDataResponseMessageProcessor(final NotificationResponseMessageSender responseMessageSender,
+            @Qualifier("domainDistributionAutomationInboundOsgpCoreResponsesMessageProcessorMap") final MessageProcessorMap messageProcessorMap) {
+        super(responseMessageSender, messageProcessorMap, MessageType.GET_DATA);
+    }
+
+    @Override
+    public void processMessage(final ObjectMessage message) throws JMSException {
+        LOGGER.debug("Processing DA GET_DATA response message");
+        this.getResponseValues(message).ifPresent(this::processResponseValues);
+    }
+
+    private Optional<ResponseValues> getResponseValues(final ObjectMessage message) {
+        final ResponseValues responseValues = new ResponseValues();
+        try {
+            responseValues.correlationUid = message.getJMSCorrelationID();
+            responseValues.messageType = message.getJMSType();
+            responseValues.organisationIdentification = message.getStringProperty(
+                    Constants.ORGANISATION_IDENTIFICATION);
+            responseValues.deviceIdentification = message.getStringProperty(Constants.DEVICE_IDENTIFICATION);
+            responseValues.responseMessage = (ResponseMessage) message.getObject();
+            responseValues.responseMessageResultType = responseValues.responseMessage.getResult();
+            responseValues.osgpException = responseValues.responseMessage.getOsgpException();
+            responseValues.payload = (String) responseValues.responseMessage.getDataObject();
+        } catch (final JMSException e) {
+            this.logJmsException(responseValues, e);
+            return Optional.empty();
+        }
+        return Optional.of(responseValues);
+    }
+
+    private void processResponseValues(final ResponseValues responseValues) {
+        try {
+            LOGGER.info("Calling application service function to handle response: {}", responseValues.messageType);
+            this.adHocManagementService.handleGetDataRequest(responseValues.payload,
```

Comment:
Consider renaming handleGetDataRequest to handleGetDataResponse.

Reply:
Done